src/metabase/public_settings/premium_features.clj

```diff
@@ -244,6 +244,22 @@
   [feature]
   (contains? (token-features) (name feature)))
 
+(defn ee-feature-error
+  "Returns an error that can be used to throw when an enterprise feature check fails."
+  [feature-name]
+  (ex-info (tru "{0} is an Enterprise feature. Please upgrade to a paid plan to use this feature." (str/capitalize feature-name))
```

Comment:
The change to use `ee-feature-error` means the feature names are not translated. I think we want to preserve the translation of those. I don't think we have an easy way to create a function that translates the arguments to the format string, so you might have to revert back to writing out the full error string each time. In that case I think it doesn't make sense to have the `assert-has-feature` function either, and instead you'll need to inline its body everywhere

Reply:
My problem with manually filling out error messages is it's inconsistent and we have to copy those messages around.

About translation, we can make the argument a translated string, and the whole message will be translated.

```
(mt/with-mock-i18n-bundles {"fr" {:messages {"{0} is an Enterprise feature. Please upgrade to a paid plan to use this feature." "{0} translated error message"
                                             "sandboxes" "Sandboxes in france"}}}
  (mt/with-temporary-setting-values [site-locale "fr"]
    (metabase.public-settings.premium-features/ee-feature-error (metabase.util.i18n/deferred-tru "sandboxes"))))
    ```
    
    try this

Reply:
I can put an assert into the `assert-has-feature` function that checks the feature name is i18n'ed

Reply:
Good idea, I'm on board with this idea and the assert is a nice touch

Reply:
I'm not necessarily opposed to this, but translating & concatenating sentence fragments is always a bit risky because it might not make grammatical sense in the target language. Different rules for word order, conjugations, subject/verb agreement, etc.

Reply:
Good point. Can't think of any languages where this case would be problematic, but it's possible